onnxruntime/test/optimizer/graph_transform_test.cc

```diff
@@ -2264,6 +2265,127 @@ TEST_F(GraphTransformationTests, FastGeluWithBiasUseGraphInputFusionTest2) {
   ASSERT_TRUE(op_to_count["FastGelu"] == 1);
 }
 
+struct BiasSoftmaxFusionTester {
+  std::shared_ptr<Model> p_model;
+  Status model_load_;
+  onnxruntime::logging::Logger* logger_;
+  onnxruntime::GraphTransformerManager graph_transformation_mgr;
+
+  bool GetAxis(const std::string op_type, const std::string name, int *axis) {
+    for (auto& node : p_model->MainGraph().Nodes()) {
+      if (node.OpType() == op_type) {
+        auto& softmax_attr = node.GetAttributes();
+        if (softmax_attr.find(name) != softmax_attr.end()) {
+          // found axis attribute
+          auto& axis_attr = softmax_attr.at(name);
+          *axis = (int)axis_attr.i();
+          return true;
+        }
+      }    
+    }
+    // not found
+    return false;
+  }
+
+  BiasSoftmaxFusionTester(
+    std::string model_uri, 
+    onnxruntime::logging::Logger* logger,
+    bool on_cuda_ = true
+  ) : logger_(logger), graph_transformation_mgr{5} {
+    model_load_ = Model::Load(model_uri, p_model, nullptr, *logger_);
+
+    // move to cuda since fusion only takes place in that case
+    if (on_cuda_) {
+      for (auto& node : p_model->MainGraph().Nodes()) {
+        node.SetExecutionProviderType(kCudaExecutionProvider);
+      }
+    }
+
+    graph_transformation_mgr.Register(
+      onnxruntime::make_unique<BiasSoftmaxFusion>(), TransformerLevel::Level2);
+  }
+
+  void TestFusionOccurs(int expected_broadcast_axis) {
+    ASSERT_TRUE(model_load_.IsOK());
```

Comment:
ASSERT_STATUS_OK

Reply:
sorry, meant for this to be a general comment. consider ASSERT_STATUS_OK for any check on a Status so we get error message output